### PR TITLE
txn: check TxnError in response

### PR DESCRIPTION
**Pull request by zhangxu19830126**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #

## What this PR does / why we need it:
The response TxnMeta needs to be checked only if TxnError is not empty, otherwise the error message returned will be incorrect

### CODE REVIEW

**Review by reusee — APPROVED** [review]

File: pkg/txn/client/operator.go [hunk 1]
```diff
@@ -505,7 +505,12 @@ func (tc *txnOperator) handleErrorResponse(resp txn.TxnResponse) error {
 	return nil
 }
 
-func (tc *txnOperator) checkResponseTxnStatusForReadWrite(txnMeta *txn.TxnMeta) error {
+func (tc *txnOperator) checkResponseTxnStatusForReadWrite(resp txn.TxnResponse) error {
+	if resp.TxnError != nil {
```
Comment (fengttt) [inline pkg/txn/client/operator.go:509]: So this is != nil, but we will return nil.    We ate this error and continue?

Comment (zhangxu19830126) [reply pkg/txn/client/operator.go:509]: handle in tc.checkTxnError

Comment (zhangxu19830126) [reply pkg/txn/client/operator.go:509]: checkResponseTxnStatus used to check if the status of TxnMeta returned by the server is correct

**Review by lni — APPROVED** [review]

**Comment by zhangxu19830126** [comment]
@Mergifyio refresh

**Comment by zhangxu19830126** [comment]
@Mergifyio refresh

**Comment by zhangxu19830126** [comment]
@Mergifyio refresh

**Pull request merged**